internal/dimovss/token_getter.go

```diff
@@ -0,0 +1,45 @@
+package dimovss
+
+import (
+	"context"
+	"time"
+
+	"github.com/DIMO-Network/model-garage/pkg/vss/convert"
+	"github.com/benthosdev/benthos/v4/public/service"
+	"golang.org/x/time/rate"
+)
+
+type LimitedTokenGetter struct {
+	tokenGetter convert.TokenIDGetter
+	limiters    map[string]*rate.Sometimes
+	logger      *service.Logger
+}
+
+func NewLimitedTokenGetter(tokenGetter convert.TokenIDGetter, logger *service.Logger) *LimitedTokenGetter {
+	return &LimitedTokenGetter{
+		tokenGetter: tokenGetter,
+		limiters:    map[string]*rate.Sometimes{},
+		logger:      logger,
+	}
+}
+
+// TokenIDFromSubject wraps a provided TokenIDGetter and logs successful queries once per day per userDevice.
+func (l *LimitedTokenGetter) TokenIDFromSubject(ctx context.Context, userDeviceID string) (uint32, error) {
+	tokenID, err := l.tokenGetter.TokenIDFromSubject(ctx, userDeviceID)
+	if err != nil {
+		return 0, err
+	}
+	limiter, ok := l.limiters[userDeviceID]
```

Comment:
I wrote a quick test, and the size of this map for 50k cars would be 2 MB.

Reply:
Is all this map access safe?

Reply:
lol no it wasn't